src/descriptor/key.rs

```diff
@@ -525,6 +525,53 @@ impl error::Error for ConversionError {
     }
 }
 
+pub trait DescriptorInnerKey {
```

Comment:
In e4d9af484b996d05db261be7406d2af78488e527:

Do you think we should add `fmt::Display` and `std::str::FromStr` bounds on this? How about `Eq`, `Hash`, etc.? (My vote is yes to all four, and maybe `fmt::Debug` too.)

cc @sanket1729 who may have opinions here.

Reply:
I added `Clone + fmt::Debug + fmt::Display + Eq + FromStr + std::hash::Hash + Ord` to match the existing derivations.